portfolio/src/main/webapp/script.js

```diff
@@ -14,21 +14,38 @@ async function handleLogin() {
 
 /**
  * Gets comments from the server.
+ * response.json() will yield a json map, where each json zmap represents a single comment with the associations found in 
+ * Comment.java, but additionally, comment[children] maps to a json array of maps recursively representing the children of the comment. 
  */
 async function getComments(limit) {
   const historyEl = document.getElementById('comments-history');
   historyEl.innerHTML = ''; //clears the table of previous comments
-  fetch('/comments?comment-limit=' + limit).then(response => response.json()).then((comments) => {
-    comments.forEach((comment) => {
-      historyEl.appendChild(createListElement(comment));
-    });
+  fetch('/comments?comment-limit=' + limit).then(response => response.json()).then((root) => {
+    renderTree(root, historyEl);
   });
 }
 
-/** Creates an <li> element containing text. */
-function createListElement(comment) {
+/** Renders the subtree rooted at root at the ul ele*/
+function renderTree(root, ele) {
+  ele.appendChild(createCommentLi(root.comment));
```

Comment:
I think we should not render the dummy node. The very first root passed into this function seems to be the dummy right?

Reply:
Yes, but the dummy node is used to provide instructions on how to comment.